### PR TITLE
Add In-Place Support for DistConv

### PR DESCRIPTION
**Pull request by fiedorowicz1**
Adds support for in-place layers with DistConv. Passing the same unit/integration tests on Lassen as the head of develop.

### CODE REVIEW

File: ci_test/common_python/test_util.py [hunk 1]
```diff
@@ -129,7 +129,7 @@ def setup_func(lbann, weekly):
             test = tools.create_tests(setup_func, file, **decorator_kwargs)[0]
             cluster = kwargs.get('cluster', 'unset')
             weekly = kwargs.get('weekly', False)
-            test(cluster, weekly, False, **decorator_kwargs)
+            test(cluster, weekly, False)
```
Comment (tbennun) [inline ci_test/common_python/test_util.py:132]: Nice

File: ci_test/unit_tests/test_unit_inplace_distconv.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        padding=1,
+        dilation=1,
+        has_bias=False,
+        parallel_strategy=ps,
+        name=f'conv_{num_dims}d'
+    )
+    z = lbann.LeakyRelu(y, negative_slope=0.2, parallel_strategy=ps,
+                        name=f'leaky_relu_{num_dims}d')
+    tester.set_loss(lbann.MeanSquaredError(z, ref))
+    return tester
```
Comment (tbennun) [inline ci_test/unit_tests/test_unit_inplace_distconv.py:57]: no newline at end of file

**Review by tbennun — APPROVED** [review]
LGTM (note minor comment on newline)

File: ci_test/unit_tests/test_unit_inplace_distconv.py (hunk 2 above)
Comment (fiedorowicz1) [reply ci_test/unit_tests/test_unit_inplace_distconv.py]: fixed

**Review by bvanessen — APPROVED** [review]
LGTM

**Pull request merged**